datafusion/src/physical_plan/hash_utils.rs

```diff
@@ -104,46 +88,11 @@ fn check_join_set_is_valid(
 
 /// Creates a schema for a join operation.
 /// The fields from the left side are first
-pub fn build_join_schema(
-    left: &Schema,
-    right: &Schema,
-    on: JoinOnRef,
-    join_type: &JoinType,
-) -> Schema {
+pub fn build_join_schema(left: &Schema, right: &Schema, join_type: &JoinType) -> Schema {
```

Comment:
same schema build simplification as the one we introduced in logical plan builder.